uwblueprint/childrens-aid-society
# Permanent Deletion Popup Modal

## Notion ticket link

[Permanent Deletion Popup Modal](https://www.notion.so/uwblueprintexecs/Permanent-Deletion-Popup-Modal-48e7696c5b7e4ee58b6486912a244cdc?pvs=4)



## Implementation description
* Made changes to button styling to add warning class so that primary button would be red
* Created new component from ModalComponent 
* Added optional fields for ModalComponent to allow to change button color and text color
* Added functionality to move state up from custom input component



## Steps to test
1. Click on Permanently Delete Button on HomePage 
<img width="769" alt="image" src="https://github.com/uwblueprint/childrens-aid-society/assets/69442773/6e5e0893-0cc0-41b9-bbb1-7c8e9f6bcddc">
2. In the input field, type "Confirm Deletion", this should turn the confirm button from a disabled state to an active state.
<img width="770" alt="image" src="https://github.com/uwblueprint/childrens-aid-society/assets/69442773/2013b15a-9413-4d85-8bdb-bafb757e5385">
 


## What should reviewers focus on?
* The modal button turning enabled with Confirm Deletion is typed in.
* If changes to ModalComponent and CustomInput components causes any other bugs.


## Checklist
